src/ess/polarization/base.py

```diff
@@ -58,116 +64,227 @@ class He3TransmissionEmptyGlass(sl.Scope[Cell, sc.DataArray], sc.DataArray):
+    wavelength bins. The time bands define intervals of different meaning, such as
+    sample runs, direct beam runs, and spin states.
     """
-    # TODO In practice, are switching times instant? Do we need to drop events that
-    # occur during switching? How is this marked in the meta data?
-    return SpinChannel()
+    # TODO
+    # Subdivide sample section into smaller intervals, or return numerator/denominator
+    # separately? The latter would complicate things when supporting different
+    # kinds of workflows, performing different kinds of normalizations.
+    # We need to be careful when subdividing and (1) exactly preserve existing bounds
+    # and (2) introduce new bounds using some heuristics that yield approximately
+    # equal time intervals (for the sample runs).
+    data = dummy_reduction(
+        time_bands=run_section.coords['time'],
+        wavelength_bands=wavelength_bands,
+    )
+    for name, log in run_section.items():
+        data.coords[name] = log.data
+    return ReducedDataByRunSectionAndWavelength(data)
+
+
+def compute_direct_beam(
+    data: sc.DataArray,
+    q_range: sc.Variable,
+    background_q_range: sc.Variable,
+) -> sc.DataArray:
+    """Compute background-subtracted direct beam function."""
+    start_db = q_range[0]
+    stop_db = q_range[-1]
+    start_bg = background_q_range[0]
+    stop_bg = background_q_range[-1]
+    # The input is binned in time and wavelength, we simply histogram without changes.
+    direct_beam = data.bins['Q', start_db:stop_db].hist()
+    background = data.bins['Q', start_bg:stop_bg].hist()
+    return direct_beam - background
+
+
+ReducedDirectBeamDataNoCell = NewType('ReducedDirectBeamDataNoCell', sc.DataArray)
+
+
+class ReducedDirectBeamData(
+    sl.ScopeTwoParams[Cell, PolarizationState, sc.DataArray], sc.DataArray
+):
+    """Direct beam data for a given cell, as a function of wavelength and time."""
+
+
+def extract_direct_beam(
+    data: ReducedDataByRunSectionAndWavelength,
+) -> ReducedDirectBeamDataNoCell:
+    """Extract direct beam without any cells from direct beam data."""
+    is_direct_beam = ~(
+        data.coords['sample_in_beam']
+        | data.coords['polarizer_in_beam']
+        | data.coords['analyzer_in_beam']
+    )
+    # We select all bins that correspond to direct-beam run sections. This preserves
+    # the separation into distinct direct beam runs, which is required later for
+    # fitting a time-decay function.
+    return ReducedDirectBeamDataNoCell(data[is_direct_beam])
+
+
+def extract_polarizer_direct_beam_polarized(
+    data: ReducedDataByRunSectionAndWavelength,
+) -> ReducedDirectBeamData[Polarizer, Polarized]:
+    """Extract run sections with polarized polarizer from direct beam data."""
+    # TODO We need all "polarized" runs, can we assume that
+    # ReducedDataByRunSectionAndWavelength does not contain any depolarized data?
+    select = (
```

Comment:
it is a measurement done only once at the very end of the cell lifetime, and not as a time-series, so I would say yes, we could seperate this from ReducedDataByRunSectionAndWavelength. But still ReducedDirectBeamData[Cell, Depolarized] has to go through the same run_reduction_workflow as the other data. That is the case, even if in the workflow above the run_reduction_workflow is not visibly connected to it, right? In base.py it looks like we use the same functions and reduction process for ReducedDirectBeamData[Cell, Depolarized]

Reply:
The question in this TODO is: Will this be recorded in the same file, i.e., do we need to do something to filter it out here?

Reply:
I see – and then yes, We should. It will probably be in the same runfile of the measurement.

Reply:
Also, both the depolarized-cell-direct-beam data, as well as the cell-out-direct-beam data will be recorded only once, i.e., be in the test-data only once. I think its best to assume that the cell-out-direct-beam-data will be recorded in the very beginning, and the depolarized in the very end.